### PR TITLE
feat(nuxt): provide `clearNuxtState` composable

### LINKED ISSUE #21293: Provide `clearNuxtState` function for `useState`
### Describe the feature

The `useState` uses `useNuxtApp().payload.state` to preserve the value after server-side rendering and share the value across all components through a unique key. If `useState` is previously used with a key and if the key is the same, it will use the cached value regardless of whether it has a different `init` method or not. It is useful to use cached data when needed. However, in some cases, the value of `useState` needs to be redeclared with a new `init` method. It seems like the only way is creating unique key every time.

It would be nice to have just like `clearNuxtData` function to clear cached state value, so we don't have to think how to make a unique key when we want to change the value of `useState`

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by jongmin4943**
### 🔗 Linked issue
#21293

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Resolves #21293

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/app/composables/state.ts [hunk 1]
```diff
@@ -35,3 +36,23 @@ export function useState <T> (...args: any): Ref<T> {
...
+      ? _allKeys.filter(keys)
+      : Array.isArray(keys)
+        ? keys.map(k => useStateKeyPrefix + k)
+        : [useStateKeyPrefix + keys]
+
+  for (const key of _keys) {
+    if (key in nuxtApp.payload.state) {
+      nuxtApp.payload.state[key] = undefined
+    }
+  }
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/state.ts:57]: slightly more performant not to do another full iteration

```suggestion
  const _keys: string[] = !keys
    ? _allKeys
    : typeof keys === 'function'
      ? _allKeys.filter(keys)
      : Array.isArray(keys) ? keys : [keys]

  for (const key of _keys) {
    if (key in nuxtApp.payload.state) {
      nuxtApp.payload.state[useStateKeyPrefix + key] = undefined
    }
  }
```

**Review by danielroe — APPROVED** [review]

**Comment by jongmin4943** [comment]
Thanks for the review! Should I write `docs` about this in this PR?

**Comment by danielroe** [comment]
Oh, yes please!

File: packages/nuxt/src/app/composables/state.ts [hunk 2]
```diff
@@ -35,3 +36,21 @@ export function useState <T> (...args: any): Ref<T> {
...
+  const nuxtApp = useNuxtApp()
+  const _allKeys = Object.keys(nuxtApp.payload.state)
+  const _keys: string[] = !keys
+    ? _allKeys
+    : typeof keys === 'function'
+      ? _allKeys.filter(keys)
+      : Array.isArray(keys) ? keys : [keys]
+
+  for (const key of _keys) {
+    if (key in nuxtApp.payload.state) {
```
Comment (jongmin4943) [inline packages/nuxt/src/app/composables/state.ts:52]: keys in `_keys` are not started with `useStateKeyPrefix("$s")`. Should the if statement be `("$s" + key in nuxtApp.payload.state)`?

Comment (danielroe) [reply packages/nuxt/src/app/composables/state.ts:52]: quite right!

File: packages/nuxt/src/app/composables/state.ts [hunk 3]
```diff
@@ -35,3 +36,21 @@ export function useState <T> (...args: any): Ref<T> {
...
+      nuxtApp.payload.state[useStateKeyPrefix + key] = undefined
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/state.ts:53]:
```suggestion
      nuxtApp.payload.state[key] = undefined
```

File: docs/3.api/3.utils/clear-nuxt-data.md [hunk 4]
```diff
@@ -12,4 +12,4 @@ clearNuxtData (keys?: string | string[] | ((key: string) => boolean)): void
 
 ## Parameters
 
-* `keys`: On or an array of keys that are used in `useAsyncData` to delete their cached data. If no keys are provided, **every data** will be invalidated.
+* `keys`: One or an array of keys that are used in `useAsyncData` to delete their cached data. If no keys are provided, **every data** will be invalidated.
```
Comment (danielroe) [inline docs/3.api/3.utils/clear-nuxt-data.md:15]:
```suggestion
* `keys`: One or an array of keys that are used in `useAsyncData` to delete their cached data. If no keys are provided, **all data** will be invalidated.
```

File: docs/3.api/3.utils/clear-nuxt-state.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+## Type
+
+```ts
+clearNuxtState (keys?: string | string[] | ((key: string) => boolean)): void
+```
+
+## Parameters
+
+* `keys`: One or an array of keys that are used in `useState` to delete their cached state. If no keys are provided, **every state** will be invalidated.
```
Comment (danielroe) [inline docs/3.api/3.utils/clear-nuxt-state.md:15]:
```suggestion
* `keys`: One or an array of keys that are used in `useState` to delete their cached state. If no keys are provided, **all state** will be invalidated.
```

**Pull request merged**